functions-app/report-psychometrician-create-download/service/psychometrician-report.service.js

```diff
@@ -16,16 +17,38 @@ const psychometricianReportUploadContainer = 'psychometricianreportupload'
 const psychometricianReportCode = 'PSR'
 const functionName = 'report-psychometrician'
 
-/**
- * Create a unique directory in the system's temp dir
- * @param prefix
- * @return {*}
- */
-async function createTmpDir (prefix) {
-  return fs.mkdtemp(`${os.tmpdir()}${path.sep}${prefix}`)
-}
-
 const psychometricianReportService = {
+  /**
+   * Create a unique directory in the system's temp dir
+   * @param prefix
+   * @return {*}
+   */
+  createTmpDir: async function createTmpDir (prefix) {
+    let tmpDir
+    if (config.PsReportTemp) {
+      this.logger(`${functionName}: config.PsReportTemp is set: ${config.PsReportTemp}`)
+      try {
+        const dirStat = await fs.stat(config.PsReportTemp)
+        if (dirStat.isDirectory()) {
+          this.logger(`${functionName}: setting tmpDir to use ${config.PsReportTemp}`)
+          // The directory exists - we can use the config
+          tmpDir = config.PsReportTemp
+        } else {
+          this.logger(`${functionName}: ERROR: not a directory, using os.tmpdir() instead`)
+          tmpDir = os.tmpdir()
+        }
+      } catch (error) {
+        // the stat failed, fallback to os tmp dir
+        this.logger(`${functionName}: ERROR: stat failed, using os.tmpdir() instead: ${error}`)
+        tmpDir = os.tmpdir()
+      }
+    } else {
+      tmpDir = os.tmpdir()
+    }
+    this.logger(`${functionName}: setting tmpDir to ${tmpDir}`)
+    return fs.mkdtemp(`${tmpDir}${path.sep}${prefix}`)
```

Comment:
Shouldn't the prefix come first?:

> fs.mkdtemp(`${prefix}${path.sep}${tmpDir}`)

Reply:
No.  Prefix is the start of the name of the directory that will be used for tmp storage, it still needs to be after the path.

Reply:
👍